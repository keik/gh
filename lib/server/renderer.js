// @flow

/**
 * Return full HTML string with React props
 *
 * @param {string} content content as HTML string
 * @param {object} props React props object for Server-Side Rendering
 * @param {string} entryName name of entrypoint
 */
export function renderFullPage(
  content: string,
  props: any,
  entryName: string,
  bust: number | string
) {
  return `
 <!DOCTYPE html>
 <html>
   <head>
     <meta charset="utf-8">
     <title>gh</title>
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.6.3/css/font-awesome.min.css">
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.7.0/styles/github.min.css">
     <link href="https://fonts.googleapis.com/css?family=Rajdhani" rel="stylesheet" />
     <link rel="stylesheet" href="/style.css?${bust}">
   </head>
   <body>
     <div id="app">${content}</div>
     <script>APP_PROPS = ${JSON.stringify(props)}</script>
     <script src="/${entryName}.js?${bust}"></script>
   </body>
 </html>
 `
}
