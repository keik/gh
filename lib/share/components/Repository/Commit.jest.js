// @flow

import * as React from 'react'
import renderer from 'react-test-renderer'

import { Commit } from './Commit'

test('<Commit /> snapshot', () => {
  const tree = renderer
    .create(
      <Commit
        commit={{
          commit: {
            author: {
              date: '$DATE',
              name: '$NAME'
            },
            sha: '$SHA',
            message: '$MESSAGE,'
          },
          files: [
            {
              additions: 10,
              blob_url: '$BLOB_URL',
              changes: 11,
              contents_url: '$CONTENTS_URL',
              deletions: 12,
              filename: '$FILENAME',
              patch: '$PATCH',
              raw_url: '$RAW_URL',
              sha: '$SHA',
              status: '$STATUS'
            }
          ],
          html_url: '$HTML_URL',
          parents: [],
          sha: '$SHA',
          stats: {
            additions: 10,
            deletions: 12,
            total: 22
          },
          url: '$URL'
        }}
        params={{ owner: '$OWNER', repo: '$REPO', sha: '$SHA' }}
        parents={[
          {
            sha: '$PARENT_SHA_1'
          },
          {
            sha: '$PARENT_SHA_2'
          }
        ]}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
