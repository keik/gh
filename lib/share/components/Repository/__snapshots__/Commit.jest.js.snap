// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Commit /> snapshot 1`] = `
<div
  className="InnerContainer-sc-1toru5t-0 eVVvlf"
>
  <div
    className="Panel-sc-1hnzj4y-0 ccxQvz"
  >
    <div
      className="Panel__Header-sc-1hnzj4y-1 gIbZlT"
    >
      <a
        className="Button-sc-1ldg16x-0 dKuqMW"
        href="/$OWNER/$REPO/tree/$SHA"
        style={
          Object {
            "float": "right",
          }
        }
      >
        Browse files
      </a>
      <div
        className="Commit__commitTitle___1uX8h"
      >
        $MESSAGE,
      </div>
      <div
        className="Commit__commitBranches___2H3Bf"
      >
        <i
          className="fa fa-code-fork"
        />
         master
      </div>
    </div>
    <div
      className="Panel__Body-sc-1hnzj4y-2 dwhEKI"
    >
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-between",
          }
        }
      >
        <div>
          <img
            alt="$NAME"
            style={
              Object {
                "background": "#ccc",
                "height": 20,
                "verticalAlign": "middle",
                "width": 20,
              }
            }
          />
           
          $NAME
           commited on
           
          <time-ago
            datetime="$DATE"
          />
        </div>
        <div
          className="Commit__sha___PySJC"
        >
          <span>
            0
             parent
          </span>
          <span>
            commit 
            $SHA
          </span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div>
      <i
        className="fa fa-file-text-o"
      />
       Showing N changed file with N addition and N deletion.
    </div>
    <div>
      <div
        className="SegmentedButtonsContainer-sc-1alpooo-0 cmuWFA"
      >
        <button
          className="Button-sc-1ldg16x-0 gIWfju"
        >
          Unified
        </button>
        <button
          className="Button-sc-1ldg16x-0 gIWfju"
        >
          Split
        </button>
      </div>
    </div>
  </div>
  <div
    id="patches"
  >
    <div
      className="Panel-sc-1hnzj4y-0 ccxQvz"
    >
      <div
        className="Panel__Header-sc-1hnzj4y-1 chmgYi"
      >
        $FILENAME
      </div>
      <div
        className="Panel__Body-sc-1hnzj4y-2 dwhEKI"
        style={
          Object {
            "padding": 0,
          }
        }
      >
        <pre
          style={
            Object {
              "margin": 0,
            }
          }
        >
          <ul
            className="Commit__lines___1WkE_"
          >
            <li>
              ...
            </li>
            <li>
              1
            </li>
            <li>
              2
            </li>
            <li>
              3
            </li>
            <li>
              4
            </li>
            <li>
              5
            </li>
            <li>
              6
            </li>
            <li>
              7
            </li>
            <li>
              8
            </li>
            <li>
              9
            </li>
            <li>
              10
            </li>
            <li>
              11
            </li>
          </ul>
          <code
            className="hljs"
            dangerouslySetInnerHTML={
              Object {
                "__html": "$PATCH",
              }
            }
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
        </pre>
      </div>
    </div>
  </div>
</div>
`;
